src/main/java/io/debezium/connector/cassandra/SchemaProcessor.java

```diff
@@ -5,24 +5,216 @@
+
+                // reinitialize the keyspace.
+                final CFMetaData cfm = oldKsm.tables.get(tableName).get();
+                final org.apache.cassandra.schema.KeyspaceMetadata newKsm = oldKsm.withSwapped(oldKsm.tables.without(tableName));
+
+                Schema.instance.unload(cfm);
+                Schema.instance.setKeyspaceMetadata(newKsm);
+
+                LOGGER.info("Removed schema for table {}", table.asCQLQuery());
+            }
+
+            @Override
+            public void onTableChanged(final TableMetadata newTableMetadata, final TableMetadata oldTableMetaData) {
+                LOGGER.debug("Detected alternation in schema of {}.{} (previous cdc = {}, current cdc = {})",
+                        newTableMetadata.getKeyspace().getName(),
+                        newTableMetadata.getName(),
+                        oldTableMetaData.getOptions().isCDC(),
+                        newTableMetadata.getOptions().isCDC());
+
+                // if it is CDC but it was not, add it to schema
+                if (newTableMetadata.getOptions().isCDC() && !oldTableMetaData.getOptions().isCDC()) {
+                    schemaHolder.addOrUpdateTableSchema(new KeyspaceTable(newTableMetadata),
+                            new SchemaHolder.KeyValueSchema(kafkaTopicPrefix, newTableMetadata, sourceInfoStructMaker));
+                }
+                else if (!newTableMetadata.getOptions().isCDC() && oldTableMetaData.getOptions().isCDC()) {
+                    // if new table is not on cdc anymore, remove it
+                    schemaHolder.removeTableSchema(new KeyspaceTable(newTableMetadata));
+                }
+                else if (newTableMetadata.getOptions().isCDC() && oldTableMetaData.getOptions().isCDC()) {
+                    // if it was cdc before and now it is too, add it, because its schema might change
+                    schemaHolder.addOrUpdateTableSchema(new KeyspaceTable(newTableMetadata),
+                            new SchemaHolder.KeyValueSchema(kafkaTopicPrefix, newTableMetadata, sourceInfoStructMaker));
+                }
+
+                // else if it was not cdc before nor now, do nothing with schema holder
+                // but add it to Cassandra for subsequent deserialization path
+
+                final CFMetaData rawNewMetadata = CFMetaData.compile(newTableMetadata.asCQLQuery(),
+                        newTableMetadata.getKeyspace().getName());
+
+                final CFMetaData oldMetadata = Schema.instance.getCFMetaData(oldTableMetaData.getKeyspace().getName(), oldTableMetaData.getName());
+
+                // we need to copy because CFMetaData.compile will generate new cfId which would not match id of old metadata
+                final CFMetaData newMetadata = rawNewMetadata.copy(oldMetadata.cfId);
+                oldMetadata.apply(newMetadata);
+
+                LOGGER.info("Updated schema for table {}", newTableMetadata.asCQLQuery());
+            }
+        };
```

Comment:
As it's difficult to write tests for this listener from your point, can you wrap the content in each function in this listener into a try-catch block and log error when it fails please? Like:

```
onTableChanged() {
       try {
              // update schemas 
       catch () {
             // LOGGER.error()
       }
 }
```